### PR TITLE
NFDIV-1348: Add checks for whether app2 email is known

### PR DESCRIPTION
**Pull request by tomxelliott**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/NFDIV-1348

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/divorce/caseworker/service/task/SendApplicationIssueNotifications.java [hunk 1]
```diff
@@ -37,7 +37,9 @@
                 }
             } else {
                 applicationIssuedNotification.sendToJointApplicant1(caseData, caseId);
-                applicationIssuedNotification.sendToJointApplicant2(caseData, caseId);
+                if (caseData.getApplication().getApplicant1KnowsApplicant2EmailAddress().toBoolean()) {
+                    applicationIssuedNotification.sendToJointApplicant2(caseData, caseId);
```
Comment (mapingo) [inline src/main/java/uk/gov/hmcts/divorce/caseworker/service/task/SendApplicationIssueNotifications.java:41]: Could getApplicant1KnowsApplicant2EmailAddress() ever return null?

Comment (tomxelliott) [reply src/main/java/uk/gov/hmcts/divorce/caseworker/service/task/SendApplicationIssueNotifications.java:41]: To get to this point they would have had to ask this question, but it never hurts to be defensive with our programming so will add a check 👍

**Review by mapingo — APPROVED** [review]

File: src/main/java/uk/gov/hmcts/divorce/caseworker/service/task/SendApplicationIssueNotifications.java [hunk 2]
```diff
@@ -37,7 +37,10 @@
                 }
             } else {
                 applicationIssuedNotification.sendToJointApplicant1(caseData, caseId);
-                applicationIssuedNotification.sendToJointApplicant2(caseData, caseId);
+                if (Objects.nonNull(caseData.getApplication().getApplicant1KnowsApplicant2EmailAddress())
```
Comment (nitinprabhu) [inline src/main/java/uk/gov/hmcts/divorce/caseworker/service/task/SendApplicationIssueNotifications.java:40]: do we need a test to ensure when this is null `applicationIssuedNotification.sendToJointApplicant2` is not executed ?

Comment (tomxelliott) [reply src/main/java/uk/gov/hmcts/divorce/caseworker/service/task/SendApplicationIssueNotifications.java:40]: @nitinprabhu - yes I did try that but due to the fact that the same mock is used for the other email execution I couldn't use verifyNoInteractions
